WinRT.Runtime/ComWrappersSupport.net5.cs

```diff
@@ -50,26 +52,32 @@ internal static unsafe InspectableInfo GetInspectableInfo(IntPtr pThis)
             return InspectableInfoTable.GetValue(_this, o => PregenerateNativeTypeInformation(o).inspectableInfo);
         }
 
-        public static object CreateRcwForComObject(IntPtr ptr)
+        public static T CreateRcwForComObject<T>(IntPtr ptr)
         {
             if (ptr == IntPtr.Zero)
             {
-                return null;
+                return default;
             }
 
+            if (typeof(T) != typeof(object))

```

Comment:
why the if check?

Reply:
Will remove as it is now redundant.  It also now check for it inside of GetRuntimeClassForTypeCreation.